pkg/puller/puller.go

```diff
@@ -71,11 +68,8 @@ func New(stateStore storage.StateStorer, topology topology.Driver, reserveState
 		syncer:       pullSync,
 		metrics:      newMetrics(),
 		logger:       logger.WithName(loggerName).Register(),
-		cursors:      make(map[string]peerCursors),
-
-		syncPeers: make([]map[string]*syncPeer, bins),
-		quit:      make(chan struct{}),
-		wg:        sync.WaitGroup{},
+		syncPeers:    make([]map[string]*syncPeer, bins),
+		quit:         make(chan struct{}),
```

Comment:
Could `quit` channel be remove in favour of global `context` (like ctx created in manage goroutine)?